eth/beacon/helpers.py

```diff
@@ -20,3 +170,100 @@ def get_active_validator_indices(dynasty: int,
         if (validator.start_dynasty <= dynasty and dynasty < validator.end_dynasty):
             o.append(index)
     return o
+
+
+#
+# Shuffling
+#
+def shuffle(lst: List[Any],
+            seed: Hash32) -> List[Any]:
+    lst_count = len(lst)
+    assert lst_count <= 16777216
+    o = [x for x in lst]
+    source = seed
+    i = 0
+    while i < lst_count:
+        source = blake(source)
+        for pos in range(0, 30, 3):
+            m = int.from_bytes(source[pos:pos + 3], 'big')
+            remaining = lst_count - i
+            if remaining == 0:
+                break
+            rand_max = 16777216 - 16777216 % remaining
+            if m < rand_max:
+                replacement_pos = (m % remaining) + i
+                o[i], o[replacement_pos] = o[replacement_pos], o[i]
+                i += 1
+    return o
+
+
+@to_tuple
+def split(lst: List[Any], number: int) -> List[Any]:
+    list_length = len(lst)
+    return [
+        lst[(list_length * i // number): (list_length * (i + 1) // number)]
+        for i in range(number)
+    ]
+
+
+@to_tuple
+def get_new_shuffling(seed: Hash32,
+                      validators: List['ValidatorRecord'],
+                      dynasty: int,
+                      crosslinking_start_shard: int,
+                      beacon_config: 'BeaconConfig') -> List[List[ShardAndCommittee]]:
```

Comment:
This function could use a docstring as well as some inline comments to explain what is happening and why.